File: solution/155/solution.py
class MinStack(object):

    def __init__(self):
        """
        initialize your data structure here.
        """
        self._data = []
        self._min = []

    def push(self, x):
        """
        :type x: int
        :rtype: void
        """
        self._data.append(x)
        if self._min and self._min[-1] < x:
            x = self._min[-1]
        self._min.append(x)

    def pop(self):
        """
        :rtype: void
        """
        self._data.pop()
        self._min.pop()

    def top(self):
        """
        :rtype: int
        """
        return self._data[-1]

    def getMin(self):
        """
        :rtype: int
        """
        return self._min[-1]

File: solution/155/solution.cpp
#include <stdio.h>
#include <stack>

class MinStack {
public:
    /** initialize your data structure here. */
    MinStack() {

    }

    void push(int x) {
        _data.push(x);
        if(!_min.empty()&&x > _min.top()){x = _min.top();}
        _min.push(x);
    }

    void pop() {
        _data.pop();
        _min.pop();
    }

    int top() {
        return _data.top();
    }

    int getMin() {
        return _min.top();
    }
private:
    std::stack<int> _data;
    std::stack<int> _min;
};